### PR TITLE
Add button has-icon state class

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Add a button state class `has-icon` which stops the button from going full width on small screens.

## QA
- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/examples/patterns/buttons/icon/
- Check that the button never goes full width at any screen size

## Details
Fixes https://github.com/vanilla-framework/vanilla-framework/issues/1922
Fixes https://github.com/vanilla-framework/vanilla-framework/issues/1902

### CODE REVIEW

**Comment by kwm14** [comment]
## Design review
- @anthonydillon I have a issue open for icons within a button but also involves icon with text too
- design issue here: https://github.com/vanilla-framework/vanilla-framework/issues/1902
- on review icon in button looks good but could you updated paddings:
   - [ ] padding left/right: `12px`

File: scss/_base_button.scss [hunk 1]
```diff
@@ -17,9 +17,9 @@
     display: inline-block;
     font-size: 1rem;
     font-weight: 300;
-    line-height: map-get($line-heights, default-text);
+    line-height: 1;
     margin-bottom: $spv-inter--scaleable + 2 * $spv-nudge-compensation;
-    padding: $spv-nudge - $px $sph-intra;
```
Comment (deadlight) [inline scss/_base_button.scss]: Won't removing this nudge break the baseline grid if the scaling changes

**Comment by kwm14** [comment]
LGTM 👍 will review Percy diffs now 💯

**Comment by kwm14** [comment]
@anthonydillon @deadlight what was the outcome of the Code-1 with this PR based on conversations this morning in @standup? 😸

File: scss/_base_button.scss (hunk 1 above)
Comment (anthonydillon) [reply scss/_base_button.scss]: Margin should to being the alignment, not padding. The scaleable adjustment is made from the `$spv-nudge-compensation` above.

Comment (deadlight) [reply scss/_base_button.scss]: Cool, makes sense

**Pull request merged**